README.md

```diff
@@ -1,4 +1,129 @@
 # xbrl-parser
+
+[![GoDoc](https://godoc.org/github.com/polygon-io/xbrl-parser?status.svg)](https://godoc.org/github.com/polygon-io/xbrl-parser)
+
 A Go library to parse xbrl documents into their facts, contexts, and units.
 
-**Waring:** This library is under development and may change dramatically!
+This library is based around the [XBRL 2.1 spec](https://www.xbrl.org/Specification/XBRL-2.1/REC-2003-12-31/XBRL-2.1-REC-2003-12-31+corrected-errata-2013-02-20.html).
+It implements support for parsing basic facts (not tuples of facts), contexts and units through the `xml.Unmarshaler` interface.
+ 
+See the package example in the godocs for how to unmarshal into the `XBRL` struct.
+
+This library supports basic validation that checks for malformed facts and broken references between facts and contexts/units (see `XBRL.Validate()`),
+but it does _not_ implement full semantic validation of XBRL documents.
+
+There are no abstractions added on-top of the XBRL data structure, which makes this library flexible and simple,
+but it also means you might have to read up a bit on how XBRL works to take full advantage of it.
+
+To give you a head start, here's some basics about XBRL:
+
+## What is XBRL?
+
+At a high level, XBRL is an XML spec intended to be the standard for digital business reporting.
+
+I can't give you a more precise definition because "digital business reporting" can mean many things, 
```

Comment:
```suggestion
We can't give you a more precise definition because "digital business reporting" can mean many things, 
```

Reply:
Maybe it's better to just remove this paragraph.

Reply:
I like the paragraph.

Reply:
I'm gonna reword some of this section, I still want to highlight the fact that digital business reporting is a broad concept